src/main/scala/org/hammerlab/guacamole/Serialization.scala

```diff
@@ -19,27 +19,78 @@
 package org.hammerlab.guacamole
 
 import com.esotericsoftware.kryo.Kryo
+import org.apache.spark.serializer.KryoRegistrator
+import org.apache.spark.storage.BroadcastBlockId
+import org.bdgenomics.adam.models.{ReferenceRegion, SequenceDictionary, SequenceRecord}
 import org.bdgenomics.adam.serialization.ADAMKryoRegistrator
+import org.bdgenomics.formats.avro.{AlignmentRecord, Genotype => BDGGenotype}
+import org.hammerlab.guacamole.commands.VariantLocus
+import org.hammerlab.guacamole.commands.jointcaller.Parameters.SomaticGenotypePolicy
+import org.hammerlab.guacamole.commands.jointcaller.annotation.{InsufficientNormal, MultiSampleAnnotations, SingleSampleAnnotations, StrandBias}
+import org.hammerlab.guacamole.commands.jointcaller.evidence.{MultiSampleMultiAlleleEvidence, MultiSampleSingleAlleleEvidence, NormalDNASingleSampleSingleAlleleEvidence, TumorDNASingleSampleSingleAlleleEvidence, TumorRNASingleSampleSingleAlleleEvidence}
+import org.hammerlab.guacamole.commands.jointcaller.{AlleleAtLocus, Input, InputCollection, Parameters}
 import org.hammerlab.guacamole.distributed.TaskPosition
+import org.hammerlab.guacamole.loci.LociMap.SimpleRange
 import org.hammerlab.guacamole.loci.{LociMap, LociMapLongSerializer, LociMapLongSingleContigSerializer, LociSet, LociSetSerializer, LociSetSingleContigSerializer}
-import org.hammerlab.guacamole.reads.{MappedRead, MappedReadSerializer, UnmappedRead, UnmappedReadSerializer}
+import org.hammerlab.guacamole.pileup.{Pileup, PileupElement}
+import org.hammerlab.guacamole.reads.{MappedRead, MappedReadSerializer, MateAlignmentProperties, PairedMappedRead, PairedRead, Read, UnmappedRead, UnmappedReadSerializer}
+import org.hammerlab.guacamole.reference.ReferenceBroadcast.MapBackedReferenceSequence
 import org.hammerlab.guacamole.variants._
 
-class GuacamoleKryoRegistrator extends ADAMKryoRegistrator {
-  override def registerClasses(kryo: Kryo) {
-    // For debugging, it can be helpful to uncomment the following line. This will raise an error if we try to serialize
-    // anything without a registered Kryo serializer.
-    // kryo.setRegistrationRequired(true)
+import scala.collection.mutable.ArrayBuffer
 
-    // This allows us to serialize object graphs with cycles. It should default to true, so kind of strange that we have
-    // to set it, but without this line we see infinite recursion.
-    kryo.setReferences(true)
+class GuacamoleKryoRegistrator extends KryoRegistrator {
+  override def registerClasses(kryo: Kryo) {
 
     // Register ADAM serializers.
-    super.registerClasses(kryo)
+    new ADAMKryoRegistrator().registerClasses(kryo)
+
+    kryo.register(classOf[Array[BDGGenotype]])
+    kryo.register(classOf[SequenceDictionary])
+    kryo.register(classOf[SequenceRecord])
+
+    kryo.register(classOf[PileupElement])
```

Comment:
Yes, and that's how I came up with many of the registrations here. The problem is that you generally only find out about one class per run (the first one that Kryo hits fails the stage), so it's very tedious to do dozens of runs to unearth all of these.